be/src/exec/pipeline/pipeline_builder.cpp

```diff
@@ -174,6 +179,55 @@ bool PipelineBuilderContext::need_local_shuffle(OpFactories ops) const {
     return down_cast<SourceOperatorFactory*>(ops[0].get())->need_local_shuffle();
 }
 
+bool PipelineBuilderContext::should_interpolate_cache_operator(OpFactoryPtr& source_op, int32_t plan_node_id) {
+    if (!_fragment_context->enable_cache()) {
+        return false;
+    }
+    const auto& cache_param = _fragment_context->cache_param();
+    if (cache_param.plan_node_id != plan_node_id) {
+        return false;
+    }
+    return dynamic_cast<pipeline::OlapScanOperatorFactory*>(source_op.get()) != nullptr;
+}
+
+OpFactories PipelineBuilderContext::interpolate_cache_operator(
+        OpFactories& upstream_pipeline, OpFactories& downstream_pipeline,
+        std::function<std::tuple<OpFactoryPtr, SourceOperatorFactoryPtr>(bool)> merge_operators_generator) {
+    DCHECK(should_interpolate_cache_operator(upstream_pipeline[0], downstream_pipeline[0]->plan_node_id()));
+
+    const auto& cache_param = _fragment_context->cache_param();
+
+    auto sink_op = upstream_pipeline.back();
+    upstream_pipeline.pop_back();
+    auto source_op = downstream_pipeline.front();
+    downstream_pipeline.clear();
+
+    auto dop = down_cast<SourceOperatorFactory*>(source_op.get())->degree_of_parallelism();
+    auto conjugate_op = std::make_shared<cache::ConjugateOperatorFactory>(sink_op, source_op);
+    upstream_pipeline.push_back(std::move(conjugate_op));
+
+    auto last_ml_op_idx = upstream_pipeline.size() - 1;
+    for (auto i = 1; i < upstream_pipeline.size(); ++i) {
+        auto& op = upstream_pipeline[i];
+        auto ml_op = std::make_shared<cache::MultilaneOperatorFactory>(next_operator_id(), op, cache_param.num_lanes);
+        // only last multilane operator in the pipeline driver can work in passthrough mode
+        auto can_passthrough = i == last_ml_op_idx;
+        ml_op->set_can_passthrough(can_passthrough);
```

Comment:
It's better `can_passthrough ` as a parameter of the constructor ?

Reply:
Only the `conjugate_op ` (StreamingAggregateSink/SourceOp) can be passthrough mode?

Reply:
Not right, can_passthrough should be true for the operator that precedes cache_operator immediately. because only this operator is computation-intensive, so its input chunk must passthrough this operator if its computation imposes an unacceptable performance penalty on cache mechanism. I will can some comment